terracotta/server/rgb.py

```diff
@@ -29,25 +29,34 @@ class Meta:
     g = fields.String(required=True, description="Key value for green band")
     b = fields.String(required=True, description="Key value for blue band")
     r_range = fields.List(
-        fields.Number(allow_none=True),
+        fields.String(allow_none=True, validate=validate.Regexp("^p?(\d*\.)?\d+$")),
```

Comment:
This is not the correct format.

Previously, this accepted any number, which includes scientific notation like `1.3E-2`. Also, percentiles must be ints between 0 and 100 and cannot be float numbers.

To prevent regressions I would advocate for still using `fields.Number` under the hood, with some additional logic to detect the percentile case. A [custom field](https://marshmallow.readthedocs.io/en/stable/custom_fields.html) class should do, similar to [what is described here](https://stackoverflow.com/questions/61614546/python-marshmallow-field-can-be-two-different-types).